homeassistant/components/hunterdouglas_powerview/shade_data.py

```diff
@@ -1,59 +1,25 @@
 """Shade data for the Hunter Douglas PowerView integration."""
 from __future__ import annotations
 
-from collections.abc import Iterable
-from dataclasses import dataclass
 import logging
 from typing import Any
 
-from aiopvapi.helpers.constants import (
-    ATTR_ID,
-    ATTR_POSITION1,
-    ATTR_POSITION2,
-    ATTR_POSITION_DATA,
-    ATTR_POSKIND1,
-    ATTR_POSKIND2,
-    ATTR_SHADE,
-)
-from aiopvapi.resources.shade import MIN_POSITION
-
-from .const import POS_KIND_PRIMARY, POS_KIND_SECONDARY, POS_KIND_VANE
-from .util import async_map_data_by_id
+from aiopvapi.resources.model import PowerviewData
+from aiopvapi.resources.shade import BaseShade, ShadePosition
 
-POSITIONS = ((ATTR_POSITION1, ATTR_POSKIND1), (ATTR_POSITION2, ATTR_POSKIND2))
+from .util import async_map_data_by_id
 
 _LOGGER = logging.getLogger(__name__)
 
```

Comment:
Not for this PR, but at some point we need to move most of this into the library and out of HA (likely when we think about the home automation post back support)